packages/app/src/runner/events/capture-protocol.ts

```diff
@@ -30,4 +30,10 @@ export const addCaptureProtocolListeners = (Cypress) => {
   Cypress.on('test:after:run', (attributes) => {
     Cypress.backend('protocol:test:after:run', attributes)
   })
+
+  Cypress.on('url:changed', (url) => {
+    const timestamp = performance.timeOrigin + performance.now()
+
+    Cypress.backend('protocol:url:changed', { url, timestamp })
```

Comment:
Once  #26508 merges in can you add the internal types for this?

Reply:
Yep, it will fail once the other PR merges.

Reply:
Added type [df97ba3](https://github.com/cypress-io/cypress/pull/26519/commits/df97ba3feb45fef9d9b5184e8d36d1f30a357596).